.github/workflows/tests.yml

```diff
@@ -408,3 +408,28 @@ jobs:
         with:
           name: ember-exam-execution-${{ matrix.browser }}-${{ hashFiles('./app/assets/javascripts/discourse/test-execution-*.json') }}
           path: ./app/assets/javascripts/discourse/test-execution-*.json
+
+  merge:
+    if: github.repository == 'discourse/discourse' && github.ref == 'refs/heads/main'
+    runs-on: debian-12
+    needs: build
+    steps:
+      - name: Merge Artifacts
+        uses: actions/upload-artifact/merge@v4
+        with:
+          name: failed-system-test-screenshots
+          pattern: failed-system-test-screenshots-*
+          delete-merged: true
+          separate-directories: false
+        # Don't fail the job if there aren't any artifacts to merge.
+        continue-on-error: true
```

Comment:
There is a feature request for [actions/upload-artifact/merge](https://github.com/actions/upload-artifact/issues/520) to support `if-no-files-found` option but there is currently no traction. 

Just continue with the step for now. Not the end of the work if we fail to upload screenshots or if flaky test reports go missing.